### DISCUSSION TITLE
End of Response

### DISCUSSION THREAD

**Discussion by SummerGram** [curl/curl#7201] [Q&A] [answered]
How does curl know the end of the response?

**Comment by bagder** [comment]
Are you asking how HTTP/1.1 works?

**Comment by jay** [comment]
Either the server lets us know (eg HTTP [Content-Length](https://datatracker.ietf.org/doc/html/rfc7230#section-3.3.2)) or closes the connection.

**Comment by SummerGram** [comment]
I made a TCP socket for fun. When the message is received, recv function will pause till it reaches the timeout.
But curl does not have this issue.

How does it work?

@jay 
No matter what content types are, the server will let us know the length?

**Comment by dfandrich** [comment, marked as answer]
It's all described in the RFC; that's the definitive source for questions about
HTTP: https://datatracker.ietf.org/doc/html/rfc7230#section-3.3.3
